### PR TITLE
Big Karp and Tiny Rat Filters

### PR DESCRIPTION
**Pull request by hammydown4325**
Allows you to filter the magikarp and rats on the map to show only ones that will count towards the medal. This is an add-on filter which means its adds on to the other filters on the map. Please test and provide feedback. Thanks!

### CODE REVIEW

**Comment by evilmoses** [comment]
Testing this now. Magikarp weight seems incorrect?

I am led to believe the karp weight criteria should be > 13.13 yet the below is showing on my map with the filter on.

capture.

Other than that, seems to be perfect, thank you so much for doing it :)

File: lib/Monocle.php [hunk 1]
```diff
@@ -31,6 +31,15 @@ public function get_active($eids, $minIv, $minLevel, $exMinIv, $swLat, $swLng, $
             $params[':oneLng'] = $oNeLng;
         }
         if (count($eids)) {
+            $tmpSQL = '';
+            if(!empty($tinyRat) && $tinyRat === 'true' && ($key = array_search("19", $eids)) === false){
```
Comment (CalamityJames) [inline lib/Monocle.php]: Would `in_array` be more appropriate here? I think it's marginally quicker if all you need to do is verify if something is in an array?

File: lib/Monocle.php [hunk 2]
```diff
@@ -32,11 +32,11 @@ public function get_active($eids, $minIv, $minLevel, $exMinIv, $bigKarp, $tinyRa
         }
         if (count($eids)) {
             $tmpSQL = '';
-            if(!empty($tinyRat) && $tinyRat === 'true' && ($key = array_search("19", $eids)) === false){
+            if (!empty($tinyRat) && $tinyRat === 'true' && ($key = array_search("19", $eids)) === false) {
```
Comment (CalamityJames) [inline lib/Monocle.php]: Would `in_array` be more appropriate here? I think it's marginally quicker if all you need to do is verify if something is in an array?

Comment (CalamityJames) [reply lib/Monocle.php]: It seems the difference is much more negligible in newer versions of PHP. Probably doesn't make much difference :)

**Pull request merged**